vertx-mysql-client/src/main/java/io/vertx/mysqlclient/impl/MySQLDatabaseMetaData.java

```diff
@@ -0,0 +1,47 @@
+package io.vertx.mysqlclient.impl;
+
+import io.vertx.sqlclient.spi.DatabaseMetaData;
+
+public class MySQLDatabaseMetaData implements DatabaseMetaData {
+  
+  private final int majorVersion;
+  private final int minorVersion;
+  private final int microVersion;
+  
+  public MySQLDatabaseMetaData(String serverVersion) {
+    // we assume the server version follows ${major}.${minor}.${micro} in https://dev.mysql.com/doc/refman/8.0/en/which-version.html
+    String[] versionNumbers = serverVersion.split("\\.");
+    majorVersion = Integer.parseInt(versionNumbers[0]);
+    minorVersion = Integer.parseInt(versionNumbers[1]);
+    // we should truncate the possible suffixes here
+    String releaseVersion = versionNumbers[2];
+    int indexOfFirstSeparator = releaseVersion.indexOf("-");
+    if (indexOfFirstSeparator != -1) {
+      // handle unstable release suffixes
+      String releaseNumberString = releaseVersion.substring(0, indexOfFirstSeparator);
+      microVersion = Integer.parseInt(releaseNumberString);
+    } else {
+      microVersion = Integer.parseInt(versionNumbers[2]);
+    }
+  }
+
+  @Override
+  public String getDatabaseProductName() {
+    return "MySQL"; // TODO: Should this return MariaDB sometimes?
```

Comment:
I think I need to rewrite the parser sometime later as it does not consider the situations of different database version semantics.

Reply:
Ok, at the moment it didn't seem like there was any MariaDB-specific code in the MySQL driver, aside from URL parsing.